### PR TITLE
[AIRFLOW-6693] Make page titles distinguishable from another

### PR DESCRIPTION
**Pull request by baecher**
Improve navigation in the browser history and when having many open
Airflow tabs in the browser. This patch includes the DAG name in the
title (where applicable) and puts it in front so that tabs can be
told apart even when having many tabs open. Likewise, if the index page
is filtered, the search term is added in front.

Before:
2020-01-29-095730_1916x1041_scrot

After:
2020-01-31-145742_3840x1080_scrot

---
Issue link: [AIRFLOW-6693](https://issues.apache.org/jira/browse/AIRFLOW-6693)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/www/templates/airflow/code.html [hunk 1]
```diff
@@ -19,9 +19,7 @@
 
 {% extends "airflow/master.html" %}
 
-{% block title %}
-    {{ title }}
-{% endblock %}
+{% block page_title %}{{ dag.dag_id }} - code - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/code.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - Code - Airflow{% endblock %}
```

File: airflow/www/templates/airflow/dag_code.html [hunk 2]
```diff
@@ -18,7 +18,8 @@
 #}
 
 {% extends "airflow/dag.html" %}
-{% block title %}Airflow - DAGs{% endblock %}
+
+{% block page_title %}{{ dag.dag_id }} - code - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/dag_code.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - Code - Airflow{% endblock %}
```

File: airflow/www/templates/airflow/dag_details.html [hunk 3]
```diff
@@ -19,9 +19,7 @@
 
 {% extends "airflow/dag.html" %}
 
-{% block title %}
-    {{ title }}
-{% endblock %}
+{% block page_title %}{{ dag.dag_id }} - details - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/dag_details.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - DAG Details - Airflow{% endblock %}
```

File: airflow/www/templates/airflow/duration_chart.html [hunk 4]
```diff
@@ -18,7 +18,7 @@
 #}
 
 {% extends "airflow/dag.html" %}
-{% block title %}Airflow - DAGs{% endblock %}
+{% block page_title %}{{ dag.dag_id }} - duration - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/duration_chart.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - Duration chart - Airflow{% endblock %}
```

File: airflow/www/templates/airflow/graph.html [hunk 5]
```diff
@@ -19,7 +19,7 @@
 
 {% extends "airflow/dag.html" %}
 
-{% block title %}Airflow - DAGs{% endblock %}
+{% block page_title %}{{ dag.dag_id }} - graph - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/graph.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - Graph - Airflow{% endblock %}
```

File: airflow/www/templates/airflow/tree.html [hunk 6]
```diff
@@ -18,7 +18,7 @@
 #}
 
 {% extends "airflow/dag.html" %}
-{% block title %}Airflow - DAGs{% endblock %}
+{% block page_title %}{{ dag.dag_id }} - tree - Airflow{% endblock %}
```
Comment (mik-laj) [inline airflow/www/templates/airflow/tree.html]:
```suggestion
{% block page_title %}{{ dag.dag_id }} - Tree - Airflow{% endblock %}
```

File: airflow/www/views.py [hunk 7]
```diff
@@ -1709,7 +1709,8 @@ def tries(self, session=None):
             demo_mode=conf.getboolean('webserver', 'demo_mode'),
             root=root,
             form=form,
-            chart=chart.htmlcontent
+            chart=chart.htmlcontent,
+            tab_title='tries',
```
Comment (mik-laj) [inline airflow/www/views.py]:
```suggestion
            tab_title='Tries',
```

File: airflow/www/views.py [hunk 8]
```diff
@@ -1788,6 +1789,7 @@ def landing_times(self, session=None):
             demo_mode=conf.getboolean('webserver', 'demo_mode'),
             root=root,
             form=form,
+            tab_title='landing times',
```
Comment (mik-laj) [inline airflow/www/views.py]:
```suggestion
            tab_title='Landing times',
```

**Comment by potiuk** [comment]
Please rebase to latest master @baecher her  - we had a failing master drama and there is much higher chance the build will succeed. We have still some intermittent errors, but there should be less often.

**Pull request merged**